cpp/modmesh/buffer/pymod/wrap_SimpleArray.cpp

```diff
@@ -75,15 +75,18 @@ class MODMESH_PYTHON_WRAPPER_VISIBILITY WrapSimpleArray
                             throw std::runtime_error("dtype mismatch");
                         }
                         modmesh::detail::shape_type shape;
+                        modmesh::detail::shape_type stride;
+                        constexpr size_t itemsize = wrapped_type::itemsize();
                         for (ssize_t i = 0; i < arr_in.ndim(); ++i)
                         {
                             shape.push_back(arr_in.shape(i));
+                            stride.push_back(arr_in.strides(i) / itemsize);
                         }
                         std::shared_ptr<ConcreteBuffer> const buffer = ConcreteBuffer::construct(
                             arr_in.nbytes(),
                             arr_in.mutable_data(),
                             std::make_unique<ConcreteBufferNdarrayRemover>(arr_in));
-                        return wrapped_type(shape, buffer);
+                        return wrapped_type(shape, stride, buffer);
```

Comment:
Why should the wrapper check if the input array is contiguous?

Reply:
Because only the wrapper sees the `PyObject` of the incoming `nearray`, on which the contiguous bit is attached available.  When the buffer pointer is passed in the `SimpleArray` C++ code, no information of memory contiguity is available.  The constructor can only trust the shape and stride passed in blindly.

Checking for contiguity in the wrapper is more robust.